src/common/AdaptiveStream.cpp

```diff
@@ -1212,17 +1238,22 @@ size_t adaptive::AdaptiveStream::getSegmentPos()
 
 bool AdaptiveStream::waitingForSegment(bool checkTime) const
 {
+  if (current_adp_->GetStreamType() == StreamType::SUBTITLE)
+    return false;
+
   if ((m_tree->HasManifestUpdates() || m_tree->HasManifestUpdatesSegs()) && state_ == RUNNING)
   {
     std::lock_guard<adaptive::AdaptiveTree::TreeUpdateThread> lckUpdTree(m_tree->GetTreeUpdMutex());
 
     if (current_rep_ && current_rep_->IsWaitForSegment())
     {
-      return !checkTime ||
-             (current_adp_->GetStreamType() != StreamType::VIDEO &&
-              current_adp_->GetStreamType() != StreamType::AUDIO) ||
-             SecondsSinceUpdate() < 1;
+      return !checkTime || (current_adp_->GetStreamType() != StreamType::VIDEO &&
+                            current_adp_->GetStreamType() != StreamType::AUDIO);
     }
+
+    // We can fall here when we are waiting for new segments, but the download it not started yet
+    // so return true only when checkTime is false, to avoid trigger EOS on sample reader and stop playback
+    return !checkTime;
```

Comment:
This is the main cause because the Pluto TV stream was stopping playback

Reply:
Good find!